src/moma_expose.cpp

```diff
@@ -174,12 +174,13 @@ Rcpp::List cpp_multirank_BIC_grid_search(
     double EPS_inner,
     long MAX_ITER_inner,
     std::string solver,
-    int selection_criterion_alpha_u  = 0,  // 0 means grid, 1 means BIC search
-    int selection_criterion_alpha_v  = 0,
-    int selection_criterion_lambda_u = 0,
-    int selection_criterion_lambda_v = 0,
-    int max_bic_iter                 = 5,
-    int rank                         = 1)
+    int deflation_scheme       = 1,  // Defaults to 1 = PCA_Hotelling
```

Comment:
Can we use `DeflationScheme::PCA_Hotelling` here instead of hard coding 1? It would make the code a bit easier to read. (Might need to change the type on `deflation_scheme`.)

Reply:
Can Rcpp handle exporting a function with an enum-type argument?

Reply:
Hmmm.... it looks like the answer is "sort of." The following works, but isn't totally type-safe. We could probably do a more general solution in the future. Issue? 

```
#include "Rcpp.h"

enum DeflationMethod {
  PCA = 0, 
  PLS = 1,  
  CCA = 2
};

namespace Rcpp {
  SEXP wrap(DeflationMethod dm){
    Rcpp::IntegerVector dm_int = Rcpp::wrap(static_cast<int>(dm));
    dm_int.attr("class") = "DeflationMethod"; 
    return dm_int;
  }

  template <> DeflationMethod as(SEXP dm_sexp){
    Rcpp::IntegerVector dm_iv(dm_sexp);
    int dm_int = dm_iv(0); 
    DeflationMethod dm = static_cast<DeflationMethod>(dm_int);
    return dm; 
  }
}

// [[Rcpp::plugins(cpp11)]]
// [[Rcpp::export]]
DeflationMethod make_pls(){
  DeflationMethod pls = DeflationMethod::PLS;
  return pls;
}

// [[Rcpp::export]]
void take_pls(DeflationMethod x){
  Rcpp::Rcout << " x = " << x << std::endl; 
}
```

Reply:
See commits "Enum-fy selection scheme encoding on R side", and "Enum-fy selection scheme encoding on C++ side".

Reply:
#60